hpc_provisioner/src/hpc_provisioner/handlers.py

```
@@ -31,50 +53,106 @@
                 logger.debug("GET version")
-                return response_text(text=version("hpc_provisioner"))
+                response = response_text(text=version("hpc_provisioner"))
+            else:
+                response = response_text(f"Unclear what to do with GET {event['path']}", code=400)
         elif event["httpMethod"] == "POST":
-            logger.debug("POST pcluster")
-            return pcluster_create_request_handler(event, _context)
+            logger.debug(
+                f"POST with path {event['path']} startswith /hpc-provisioner/dra: {event['path'].startswith('/hpc-provisioner/dra')}"
+            )
+            if event["path"] == "/hpc-provisioner/pcluster":
+                logger.debug("POST pcluster")
+                response = pcluster_create_request_handler(event, _context)
+            elif event["path"].startswith("/hpc-provisioner/dra"):
+                logger.debug("POST DRA")
+                response = dra_check_handler(event, _context)
+            else:
+                response = response_text(f"Unclear what to do with POST {event['path']}", code=400)
         elif event["httpMethod"] == "DELETE":
             logger.debug("DELETE pcluster")
-            return pcluster_delete_handler(event, _context)
+            response = pcluster_delete_handler(event, _context)
         else:
-            return response_text(f"{event['httpMethod']} not supported", code=400)
+            response = response_text(f"{event['httpMethod']} not supported", code=400)
+    else:
+        response = response_text(f"Unclear what to do with event {event}", code=400)
+    logger.debug(f"Response: {response}")
+    return response
+
+
+def dra_check_handler(event, _context=None):
+    """
+    1. Check which clusters are pending (provisioning_launched=False) creation and have include_lustre=True
+    2. For each of them:
+        check whether any DRAs are still pending
+        if not: call do_cluster_create
```

Just a personal opinion: if I understand correctly, the logic is to create the filesystems first and, once successful, we create the pcluster. I'd swap the order of the operations to promote a "fail fast" approach: if the cluster creation (which in theory is faster/lighter than the filesystem mounting) fails for any reason , then, we fail "fast" and don't have to wait for the slower/heavier FS mounting process.